core/src/test/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAOSpec.scala

```diff
@@ -0,0 +1,115 @@
+package org.broadinstitute.dsde.rawls.billing
+
+import org.broadinstitute.dsde.rawls.config.{AzureConfig, MultiCloudWorkspaceConfig}
+import org.broadinstitute.dsde.rawls.dataaccess.SamDAO
+import org.broadinstitute.dsde.rawls.dataaccess.slick.TestDriverComponent
+import org.broadinstitute.dsde.rawls.model.{CreationStatuses, RawlsBillingProject, RawlsBillingProjectName, SamBillingProjectActions, SamBillingProjectRoles, SamResourceAction, SamResourceTypeNames, SamRolesAndActions, SamUserResource, AzureManagedAppCoordinates}
+import org.mockito.Mockito.when
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers._
+import org.scalatestplus.mockito.MockitoSugar
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
+
+class BillingProfileManagerDAOSpec extends AnyFlatSpec with TestDriverComponent with MockitoSugar {
+  val azConfig: AzureConfig = AzureConfig(
+    "fake-sp-id",
+    "fake-tenant-id",
+    "fake-sub-id",
+    "fake-mrg-id",
+    "fake-bp-name",
+    "fake-alpha-feature-group"
+  )
+
+  behavior of "listBillingProfiles"
+
+  it should "return billing profiles to which the user has access" in {
+    val samDAO: SamDAO = mock[SamDAO]
+    when(samDAO.userHasAction(SamResourceTypeNames.managedGroup, azConfig.alphaFeatureGroup,
+      SamResourceAction("use"),
+      userInfo
+    )).thenReturn(Future.successful(true))
+    val bpSamResource = SamUserResource(azConfig.billingProjectName,
+      SamRolesAndActions(
+        Set(SamBillingProjectRoles.owner),
+        Set(SamBillingProjectActions.createWorkspace)
+      ),
+      SamRolesAndActions(Set.empty, Set.empty),
+      SamRolesAndActions(Set.empty, Set.empty),
+      Set.empty,
+      Set.empty
+    )
+    when(samDAO.listUserResources(SamResourceTypeNames.billingProject, userInfo)).thenReturn(
```

Comment:
can we add a Google billing project to the return value to make sure the filter is working correctly?